src/lib.rs

```diff
@@ -1013,18 +1021,26 @@ impl Node {
 		}
 		let runtime = rt_lock.as_ref().unwrap();
 
-		let cur_balance = self.wallet.get_balance()?;
-		if cur_balance.get_spendable() < channel_amount_sats {
-			log_error!(self.logger, "Unable to create channel due to insufficient funds.");
-			return Err(Error::InsufficientFunds);
-		}
-
 		let peer_info = PeerInfo { node_id, address };
 
 		let con_node_id = peer_info.node_id;
 		let con_addr = peer_info.address.clone();
 		let con_cm = Arc::clone(&self.connection_manager);
 
+		let cur_anchor_reserve_sats =
+			total_anchor_channels_reserve_sats(&self.channel_manager, &self.config);
+		let spendable_amount_sats =
+			self.wallet.get_balances(cur_anchor_reserve_sats).map(|(_, s)| s).unwrap_or(0);
```

Comment:
Yeah, ideally `Wallet` would just be aware of the anchors reserve (instead of passing it) and return an adjusted `Balance`, though that involves introducing dependencies to the `Wallet`.  Even if it were aware, the reserve concept can't be applied to one of the `Balance` components without affecting the total, unfortunately.

What about defining our own internal `Balance` as:

```rust
struct Balance {
	pub anchor_reserve_sats: u64,
	pub spendable_amount_sats: u64,
	pub total_onchain_balance_sats: u64,
}
```

Where `Wallet::get_balances` takes `ChannelManager` and `Config` to compute the anchor reserve internally instead of computing it externally at each call site and passing it in? That would keep the calculations in one place, including the one that adjusts the reserve based on the total balance.